clc/modules/object-storage-common/src/main/java/com/eucalyptus/objectstorage/msgs/ObjectStorage.groovy

```diff
@@ -66,8 +69,18 @@ import edu.ucsb.eucalyptus.msgs.StreamedBaseMessage
 @ComponentMessage(ObjectStorage.class)
 public class ObjectStorageResponseType extends ObjectStorageRequestType {
   HttpResponseStatus status; //Most should be 200-ok, but for deletes etc it may be 204-No Content
+  protected String bucketUuid;
 
   def ObjectStorageResponseType() {}
+
+  public String getBucketUuid() {
+    return bucketUuid;
+  }
+
+  public void setBucketUuid(String bucketUuid) {
+    this.bucketUuid = bucketUuid;
+  }
+
```

Comment:
These methods in ObjectStorageGateway set the bucket name to the bucket UUID: (all others set it to the bucket name)
		○ getObject()
		○ getObjectExtended()
		○ generateNotModifiedResponse()
		○ headObject()
		○ initiateMultipartUpload() (but saves originalBucket name)
		○ abortMultipartUpload()
	Resolved by adding a bucket name field to the response and getting that name from the bucket object, not the request.